File: frontend/src/types/project.ts
export interface Project {
  id: number
  owner_id: number
  title: string
  description: string
  format: 'remote' | 'office' | 'hybrid'
  payment_type: 'volunteer' | 'paid' | 'equity'
  status: 'draft' | 'open' | 'closed' | 'completed' | 'archived'
  skills: Skill[]
  members?: ProjectMember[]
  owner?: ProjectOwner
  score?: number
  created_at: string
  updated_at: string
}

export interface ProjectOwner {
  id: number
  full_name: string
  avatar_url?: string
}

export interface ProjectMember {
  id: number
  role: string
  joined_at?: string
  is_active: boolean
}
export interface Skill {
  id: number
  name: string
}

export interface ProjectCreate {
  title: string
  description: string
  format: 'remote' | 'office' | 'hybrid'
  payment_type: 'volunteer' | 'paid' | 'equity'
  skill_ids: number[]
  status?: 'draft' | 'open' | 'closed' | 'archived'
}

export interface ProjectUpdate {
  title?: string
  description?: string
  format?: 'remote' | 'office' | 'hybrid'
  payment_type?: 'volunteer' | 'paid' | 'equity'
  status?: 'draft' | 'open' | 'closed' | 'completed' | 'archived'
  skill_ids?: number[]
}

File: backend/app/schemas/project.py
from datetime import datetime
from typing import Optional
from enum import Enum
from pydantic import BaseModel, Field
from app.schemas.skill import SkillResponse
from app.models.project import ProjectStatus



class MemberUserResponse(BaseModel):
    id: int
    full_name: str
    avatar_file_id: Optional[int] = None

    model_config = {"from_attributes": True}
class ProjectOwnerResponse(BaseModel):
    id: int
    full_name: str
    avatar_file_id: Optional[int] = None
    
    model_config = {"from_attributes": True}
class ProjectMemberResponse(BaseModel):
    user_id: int
    role: str
    joined_at: Optional[datetime] = None
    is_active: bool = True
    user: MemberUserResponse
    
    model_config = {"from_attributes": True}


class ProjectResponse(BaseModel):
    id: int
    owner_id: int
    title: str
    description: str
    format: str
    payment_type: str
    status: ProjectStatus
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
    skills: list[SkillResponse] = []
    members: list[ProjectMemberResponse] = []
    owner: Optional[ProjectOwnerResponse]
    model_config = {"from_attributes": True}


class ProjectCreateRequest(BaseModel):
    title: str = Field(..., min_length=1, max_length=200)
    description: str = Field(..., min_length=1, max_length=5000)
    format: str = Field(default="remote", pattern=r"^(remote|office|hybrid)$")
    payment_type: str = Field(default="volunteer", pattern=r"^(volunteer|paid|equity)$")
    status: ProjectStatus = Field(
        default=ProjectStatus.OPEN,
        description="Статус проекта: draft/open/closed/archived"
    )
    skill_ids: list[int] = []


class ProjectUpdateRequest(BaseModel):
    title: Optional[str] = Field(None, min_length=1, max_length=200)
    description: Optional[str] = Field(None, min_length=1, max_length=5000)
    format: Optional[str] = Field(None, pattern=r"^(remote|office|hybrid)$")
    payment_type: Optional[str] = Field(None, pattern=r"^(volunteer|paid|equity)$")
    status: Optional[ProjectStatus] = Field(None, pattern=r"^(draft|open|closed|archived)$")
    skill_ids: Optional[list[int]] = None
